presto-main/src/test/java/com/facebook/presto/sql/analyzer/TestMaterializedViewQueryOptimizer.java

```diff
@@ -933,32 +1234,83 @@ public void testFilterContainmentWithMismatchStringLength()
-        expectedRewrittenSql = format("SELECT a, b FROM %s WHERE b = 'UK'", VIEW);
+        expectedRewrittenSql = format("SELECT a, b FROM %s WHERE b = 'UK'", VIEW_1);
 
-        assertOptimizedQuery(originalViewSql, baseQuerySql, expectedRewrittenSql);
+        assertOptimizedQuery(baseQuerySql, expectedRewrittenSql, originalViewSql);
     }
 
-    private void assertOptimizedQuery(String originalViewSql, String baseQuerySql, String expectedViewSql)
+    private void assertOptimizedQuery(String baseQuerySql, String expectedQueryAfterOptimization, String... existingViewQueries)
     {
-        Table viewTable = new Table(QualifiedName.of(VIEW));
-
-        Query originalViewQuery = (Query) SQL_PARSER.createStatement(originalViewSql, PARSING_OPTIONS);
-        Query baseQuery = (Query) SQL_PARSER.createStatement(baseQuerySql, PARSING_OPTIONS);
-        Query expectedViewQuery = (Query) SQL_PARSER.createStatement(expectedViewSql, PARSING_OPTIONS);
-
         transaction(transactionManager, accessControl)
                 .singleStatement()
                 .readUncommitted()
                 .readOnly()
                 .execute(CLIENT_SESSION, session -> {
-                    Query optimizedBaseToViewQuery = (Query) new MaterializedViewQueryOptimizer(metadata, session, SQL_PARSER, accessControl, domainTranslator, viewTable, originalViewQuery).rewrite(baseQuery);
+                    Query baseQuery = (Query) SQL_PARSER.createStatement(baseQuerySql, createParsingOptions(session));
+                    Query expectedViewQuery = (Query) SQL_PARSER.createStatement(expectedQueryAfterOptimization, createParsingOptions(session));
+                    Map<QualifiedObjectName, List<ConnectorMaterializedViewDefinition>> baseTableToMaterializedViewMap = new HashMap<>();
+
+                    int i = 1;
+                    for (String existingViewQuery : existingViewQueries) {
+                        QualifiedObjectName viewTableName = buildQualifiedNameFromViewQuery(session, existingViewQuery);
+                        baseTableToMaterializedViewMap.computeIfAbsent(viewTableName, (x) -> new ArrayList<>());
+                        baseTableToMaterializedViewMap.get(viewTableName).add(createStubConnectorMaterializedViewDefinition(existingViewQuery, i++));
+                    }
+
+                    Query optimizedBaseToViewQuery = new MaterializedViewQueryOptimizer(
+                            metadata,
+                            session,
+                            SQL_PARSER,
+                            accessControl,
+                            domainTranslator,
+                            baseTableToMaterializedViewMap)
+                            .rewrite(baseQuery);
                     assertEquals(optimizedBaseToViewQuery, expectedViewQuery);
                 });
     }
+
+    private QualifiedObjectName buildQualifiedNameFromViewQuery(Session session, String existingViewQuery)
```

Comment:
This is very hacky. Let send view names as map<view name, sql> from tests so we dont do this weird parsing